### PR TITLE
Add report version into details and deployments report

### LINKED ISSUE #1618: Add report version into reports
### Is your feature request related to a problem?
SEAP is building tools that consume CSV reports produced by QPC.  At some future point, we may need to upgrade the report.  Providing a version in the report will allow us to determine the version of a report.  Previous to this enhancement, the reports were not considered API.  This feature allows us to freeze features with a version.

### PR DESCRIPTION
**Pull request by kholdaway**
Closes #1618

### CODE REVIEW

File: quipucords/api/messages.py [hunk 1]
```diff
@@ -16,6 +16,8 @@
 FC_REQUIRED_ATTRIBUTE = 'Required. May not be null or empty.'
 FC_SOURCE_NAME_NOT_STR = 'Source name must be type string.'
 FC_MUST_BE_ONE_OF = 'Must be one of the following: %s'
+FC_MISSING_REPORT_VERSION = 'Missing report_version.  '\
+    'Future releases will not tolerate missing or invalid report_version.'
```
Comment (abaiken) [inline quipucords/api/messages.py]: possibly change the second sentence to `Future releases will not tolerate a missing or invalid report_version.`

Comment (kholdaway) [reply quipucords/api/messages.py]: Fixed

File: qpc/report/merge.py [hunk 2]
```diff
@@ -120,34 +97,69 @@ def _get_report_ids(self):
...
+                                print(
+                                    _(messages.REPORT_JSON_MISSING_SERVER_ID %
+                                      file))
+                                has_error = True
+                                break
+                        if not has_error:
+                            if not report_version and \
+                                    details_report.get('report_version',
+                                                       None) and \
+                                    details_report.get('report_type',
```
Comment (abaiken) [inline qpc/report/merge.py]: why are we looking up the `report_type` again here? We already saved in a variable on line 121

File: qpc/report/merge.py [hunk 3]
```diff
@@ -120,34 +97,69 @@ def _get_report_ids(self):
...
+                        details_report = json.load(lint_f)
                     except json_exception_class:
                         print(_(messages.REPORT_JSON_DIR_FILE_FAILED % file))
                         continue
-                    sources = json_data.get('sources')
+
+                    # validate report type
+                    report_type = details_report.get('report_type', None)
+                    if not report_type:
+                        print(_(messages.REPORT_MISSING_REPORT_VERSION % file))
```
Comment (abaiken) [inline qpc/report/merge.py]: It seems like we want an error message about `report_type here`, not `report_version`.

File: qpc/report/merge.py [hunk 4]
```diff
@@ -120,34 +97,69 @@ def _get_report_ids(self):
...
+                                has_error = True
+                                break
+                            if not server_id:
+                                print(
+                                    _(messages.REPORT_JSON_MISSING_SERVER_ID %
+                                      file))
+                                has_error = True
+                                break
+                        if not has_error:
+                            if not report_version and \
```
Comment (abaiken) [inline qpc/report/merge.py]: I don't understand this check. Could you explain it to me? From what I can tell, report_version can only be None at this point, and if all we are doing if this passes, is setting the report_version, why do we need this check? Ie. why can't we just do:
```report_version = details_report.get('report_version', None)```

File: qpc/messages.py [hunk 5]
```diff
@@ -167,11 +167,22 @@
                          'required to merge.'
 REPORT_INVALID_JSON_FILE = 'The file %s does not contain a valid json ' \
                            'details report.'
+REPORT_MISSING_REPORT_VERSION = 'The file %s is missing report_version.  '\
+    'Future releases will not tolerate a missing or invalid report_version.'
+REPORT_INVALID_REPORT_TYPE = '%s is not a valid report type.  '\
+    'Only details reports can be merged. Excluding from merge.'
```
Comment (abaiken) [inline qpc/messages.py]: Should we include which file is being excluded in this error message?

File: qpc/report/merge.py (hunk 3 above)
Comment (kholdaway) [reply qpc/report/merge.py]: Fixed check to be report_version

File: qpc/report/merge.py (hunk 2 above)
Comment (kholdaway) [reply qpc/report/merge.py]: Take a look again with new locals

File: qpc/report/merge.py (hunk 4 above)
Comment (kholdaway) [reply qpc/report/merge.py]: Take a look again with new locals

File: qpc/messages.py [hunk 6]
```diff
@@ -167,11 +167,23 @@
...
+    'The file %s is missing report_version.  '\
+    'Future releases will not tolerate a missing or invalid report_version.'
+REPORT_INVALID_REPORT_TYPE = 'The file %s contains invalid report type %s.  '\
+    'Only details reports can be merged. Excluding from merge.'
 REPORT_JSON_DIR_NO_FILES = \
     'No files with extension .json found in %s.'
 REPORT_VALIDATE_JSON = 'Checking files for valid json details report. %s'
-REPORT_JSON_DIR_FILE_FAILED = 'Failed: %s is not a details report.'
-REPORT_JSON_DIR_FILE_SUCCESS = 'Success: %s is a details report.'
+REPORT_JSON_DIR_FILE_FAILED = 'Failed: %s is not a details report. '\
```
Comment (abaiken) [inline qpc/messages.py]: Is there any way that we can condense the following 3 error messages?

Comment (kholdaway) [reply qpc/messages.py]: Done

**Pull request merged**